### PR TITLE
New Tile Class and whatDoYourSnakeEyesSee method

### PR DESCRIPTION
**Pull request by DHervochon**
Hasn't been tested but doesn't break anything, since nothing uses these yet.
Should be useful for cutting down time on pathfinding to food, and sets the foundation for panic mode.
Might want to move these into a classes file for Tile and parseBoard for snakeEyes.

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
+class Tile():
     """
-    Finds Snakes that are smaller than us and assigns the area around
-    their head with the Smaller_Snake_Future_Head
+    TODO: Add functionality for what kind of tile it is, ie: snakeBody, ourHead, ourBody, food, empty tile
+    This class exists so that whatDoYourSnakeEyesSee only needs to look at the associated object of a tile to see if it's been
+    visited, instead of searching the entire list of visited tiles every time. Should be usable for other things.
+
+    Attributes:
+        x (integer): the x coordinate, should line up with our other methods
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        x (int): the x coordinate, should line up with our other methods
```

File: app/main.py [hunk 2]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
+        y (integer): the y coordinate etc
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        y (int): the y coordinate etc
```

File: app/main.py [hunk 3]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
+        visited (boolean): whether or not a tile has been visited, useful for some algorithms
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        visited (bool): whether or not a tile has been visited, useful for some algorithms
```

File: app/main.py [hunk 4]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
-                            board[y][x-1] = 3
-                    if(y < converted_data['board']['width'] and y > 0):
-                        if(pathBoard[y+1][x] != -1):
-                            board[y+1][x] = 3
-                        if(pathBoard[y-1][x] != -1):
-                            board[y-1][x] = 3
+    This method finds the tiles it is possible to path to from any tiles coordinates, and the tiles that form walls around/in this area,
+    Then creates lists for this information.
+    Args:
+        pathBoard (array): integer representation of current board
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        pathBoard (list): integer representation of current board 
```

File: app/main.py [hunk 5]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
-                    if(y < converted_data['board']['width'] and y > 0):
-                        if(pathBoard[y+1][x] != -1):
-                            board[y+1][x] = 3
-                        if(pathBoard[y-1][x] != -1):
-                            board[y-1][x] = 3
+    This method finds the tiles it is possible to path to from any tiles coordinates, and the tiles that form walls around/in this area,
+    Then creates lists for this information.
+    Args:
+        pathBoard (array): integer representation of current board 
+        xPos(int): the x coordinate of the tile we want to search from
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        xPos (int): the x coordinate of the tile we want to search from
```

File: app/main.py [hunk 6]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
+        yPos(int): the y coordinate of the tile we want to search from
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        yPos (int): the y coordinate of the tile we want to search from
```

**Review by DimaMzk — APPROVED** [review]

**Review by AGuyNamedTate — CHANGES_REQUESTED** [review]
Needs to pull changes from other branch. Currently breaks because of the printBoard function being moved to another file.

**Review by AGuyNamedTate — DISMISSED** [review]

File: app/main.py [hunk 7]
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
...
+    OBJECT_ORIENTED_TODO: pass in a board object and tile object instead
+    Returns:
+        an list of two lists, the first contains all the Tile objects it is possible and viable to path to, the second 
+        contains all Tile objects that form the walls. Can compare these lists to lists of all food, for example, to get
+        just food we can path too.
+    OBJECT_ORIENTED_TODO: return board objects instead eventually maybe
+    """
+    board_width = len(pathBoard[0])
+    board_height = len(pathBoard)
+    allBoardTiles = [[None for x in range(board_width)] for y in range(board_height)]
```
Comment (AGuyNamedTate) [inline app/main.py]: Please review variable name to reflect python naming conventions (all_board_tiles)

**Review by AGuyNamedTate — CHANGES_REQUESTED** [review]
Still requires rebase

**Pull request closed without merging**